userspace/libsinsp/filterchecks.cpp

```diff
@@ -609,19 +615,36 @@ uint8_t* sinsp_filter_check_fd::extract(sinsp_evt *evt, OUT uint32_t* len, bool
 			}
 
 			scap_fd_type evt_type = m_fdinfo->m_type;
-			if(evt_type == SCAP_FD_IPV4_SOCK)
+			if(m_field_id != TYPE_CLIENTIP_NAME)
 			{
-				RETURN_EXTRACT_VAR(m_fdinfo->m_sockinfo.m_ipv4info.m_fields.m_sip);
+				if(evt_type == SCAP_FD_IPV4_SOCK)
+				{
+					RETURN_EXTRACT_VAR(m_fdinfo->m_sockinfo.m_ipv4info.m_fields.m_sip);
+				}
+				else if (evt_type == SCAP_FD_IPV6_SOCK)
+				{
+					RETURN_EXTRACT_VAR(m_fdinfo->m_sockinfo.m_ipv6info.m_fields.m_sip);
+				}
 			}
-			else if (evt_type == SCAP_FD_IPV6_SOCK)
+			else
 			{
-				RETURN_EXTRACT_VAR(m_fdinfo->m_sockinfo.m_ipv6info.m_fields.m_sip);
+				if(evt_type == SCAP_FD_IPV4_SOCK)
+				{
+					m_tstr = sinsp_dns_manager::get().name_of(&m_fdinfo->m_sockinfo.m_ipv4info.m_fields.m_sip, false);
+				}
+				else if (evt_type == SCAP_FD_IPV6_SOCK)
+				{
+					m_tstr = sinsp_dns_manager::get().name_of(&m_fdinfo->m_sockinfo.m_ipv6info.m_fields.m_sip.m_b[0], true);
+				}
+
+				RETURN_EXTRACT_STRING(m_tstr);
 			}
 		}
 
 		break;
 	case TYPE_SNET:
 	case TYPE_SERVERIP:
+	case TYPE_SERVERIP_NAME:
```

Comment:
I think these cases only share a couple of lines (fetching the fdinfo), but are otherwise unrelated. Do you want to have a separate case for the _NAME ones?

Reply:
I was trying to minimize the code duplication (already heavy in this file) but yes, in this case I suppose it's better as you are proposing. Will do.